Modules/_datetimemodule.c

```diff
@@ -1290,6 +1290,9 @@ tzinfo_from_isoformat_results(int rv, int tzoffset, int tz_useconds) {
         }
 
         PyObject *delta = new_delta(0, tzoffset, tz_useconds, 1);
+        if (delta == NULL) {
+            return NULL;
+        }
         tzinfo = new_timezone(delta, NULL);
         Py_XDECREF(delta);
```

Comment:
Seems like this should be changed to straight `Py_DECREF`.

Reply:
Indeed, I missed that. Thank you!